kobo/apps/trash_bin/tasks/attachment.py

```diff
@@ -53,3 +57,28 @@ def empty_attachment_failure(sender=None, **kwargs):
 @task_retry.connect(sender=empty_attachment)
 def empty_attachment_retry(sender=None, **kwargs):
     trash_bin_task_retry(AttachmentTrash, **kwargs)
+
+
+# @celery_app.task
+def schedule_auto_attachment_cleanup_for_users():
+    """
+    Identifies users exceeding storage limits beyond the grace period and
+    schedules a cleanup task for each.
+
+    Runs only if AUTO_DELETE_ATTACHMENTS and Stripe billing is enabled.
+    """
+    if not config.AUTO_DELETE_ATTACHMENTS or not settings.STRIPE_ENABLED:
+        return
+
+    exceeded_counters = ExceededLimitCounter.objects.filter(
+        limit_type=UsageType.STORAGE_BYTES,
+        days__gte=config.STORAGE_OVERAGE_ATTACHMENT_DELETION_GRACE_PERIOD
+    ).select_related('user')
+
+    for counter in exceeded_counters:
+        auto_delete_excess_attachments(counter.user.pk)
```

Comment:
what about using `counter.user_id` and not using `select_related('user')` if we don't need anything else to `User` model